src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTestBase.SocketsHttpHandler.cs

```diff
@@ -16,7 +16,7 @@ public abstract partial class HttpClientHandlerTestBase : FileCleanupTestBase
 
         protected virtual QuicImplementationProvider UseQuicImplementationProvider => null;
 
-        public static bool IsMsQuicSupported => QuicImplementationProviders.MsQuic.IsSupported;
+        public static bool IsMsQuicSupported => PlatformDetection.IsNotBrowser && QuicImplementationProviders.MsQuic.IsSupported;
```

Comment:
Should `QuicImplementationProviders.MsQuic.IsSupported` be updated to handle the browser case? That way, if an app tries to use that API, then it would return `false` for them?

Reply:
The problem is that QuicImplementationProviders throws PlatformNotSupportedException on AssemblyInfo level.

Reply:
What do you mean by "on AssemblyInfo level"?

Reply:
https://github.com/dotnet/runtime/commit/33d881fd04242635e3d667f93acaad433baa308f#diff-1d4ce147edca0ce81f312b96fce76ac52eb02eb6d04e660817e3b47230edc794R9

Reply:
IIUC, `QuicImplementationProviders.MsQuic` will throw PNSE. And the supported platforms are at least partially being encoded via the target frameworks in the project file. So, IMHO, instead of explicitly adding `IsNotBrowser` here, I think we should return `QuicImplementationProviders.MsQuic.IsSupported`, and if it throws PNSE, then return `false`.

Reply:
Okay